File: server/DocumentsKM/Models/ProfileClass.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DocumentsKM.Models
{
    // Вид профиля
    public class ProfileClass
    {
        [Key]
        public Int16 Id { get; set; }

        // Название
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        // Примечание
        [MaxLength(255)]
        public string Note { get; set; }
    }
}

File: client/src/model/ProfileClass.ts
interface IProfileClass {
	id: number
	name: string
	note: string
}

class ProfileClass {
	id: number
	name: string
	note: string

	constructor(obj?: IProfileClass) {
		this.id = (obj && obj.id) || 0
		this.name = (obj && obj.name) || ''
		this.note = (obj && obj.note) || ''
	}
}

export default ProfileClass
